frontend/src/lib/utils.tsx

```diff
@@ -632,12 +632,29 @@ export function dateFilterToText(
     }
     dateFrom = (dateFrom || undefined) as string | undefined
     dateTo = (dateTo || undefined) as string | undefined
+
     if (isDate.test(dateFrom || '') && isDate.test(dateTo || '')) {
         return `${dateFrom} - ${dateTo}`
     }
+
     if (dateFrom === 'dStart') {
+        // Changed to "last 24 hours" but this is backwards compatibility
         return 'Today'
-    } // Changed to "last 24 hours" but this is backwards compatibility
+    }
+
+    if (isDate.test(dateFrom || '') && !isDate.test(dateTo || '')) {
```

Comment:
Think it's worth adding a comment in the code on why this edge case happens (i.e. why could `dateFrom` be set and not `dateTo`)

Reply:
Basically you can never trust incoming data :)... and I'm not sure where it comes from, and/or where it will come from next time.